src/bptl/work_units/valid_sign/tasks.py

```diff
@@ -0,0 +1,264 @@
+
+    def format_signers(self, signers: List[dict]) -> List[dict]:
+        """
+        Formats the signer information into an array of JSON objects as needed by ValidSign.
+        """
+        formatted_signers = []
+
+        for signer in signers:
+            formatted_signers.append({"type": "SIGNER", "signers": [signer]})
+
+        return formatted_signers
+
+    def _get_documents_from_api(self) -> List[Tuple[str, bytes]]:
+        """
+        Retrieves the documents from Documenten API and returns a list of tuples with the name and the binary content
+        of each document.
+        """
+        logger.debug("Retrieving documents from Documenten API")
+
+        variables = self.task.get_variables()
+        document_urls = variables.get("documents")
+
+        documents = []
+        for document_url in document_urls:
+            # Retrieving the document
+            response = requests.get(
+                document_url,
+                auth=(settings.DOCUMENT_API_USER, settings.DOCUMENT_API_PASSWORD),
+            )
+            response.raise_for_status()
+
+            document_data = response.json()
+            # Retrieving the content of the document
+            response = requests.get(
+                document_data.get("inhoud"),
+                auth=(settings.DOCUMENT_API_USER, settings.DOCUMENT_API_PASSWORD),
+            )
+            response.raise_for_status()
+
+            documents.append((document_data.get("titel"), response.content))
+
+        return documents
+
+    def _get_signers_from_package(self, package: dict) -> List[dict]:
+        """
+        Retrieves all the roles from a ValidSign package and returns all the signers.
+        """
+        logger.debug(
+            f"Retrieving the roles from validSign package '{package.get('id')}'"
+        )
+
+        roles_url = (
+            f"{settings.VALIDSIGN_ROOT_URL}api/packages/{package.get('id')}/roles"
+        )
+        response = requests.get(roles_url, headers=self._auth_header)
+        response.raise_for_status()
+        roles = response.json().get("results")
+        # Not all the roles are signers (one of them is the account owner)
+        return [role for role in roles if role.get("type") == "SIGNER"]
+
+    def create_package(self) -> dict:
+        """
+        Creates a ValidSign package with the name specified by the process variables and adds the signers to it.
+        """
+        logger.debug("Creating ValidSign package")
+
+        variables = self.task.get_variables()
+        signers = self.format_signers(variables.get("signers"))
+        package_name = variables.get("package_name")
+
+        url = f"{settings.VALIDSIGN_ROOT_URL}api/packages"
+
+        body = {
+            "name": package_name,
+            "type": "PACKAGE",
+            "roles": signers,
+        }
+
+        response = requests.post(url, headers=self._auth_header, data=json.dumps(body))
```

Comment:
zgw consumers client abstracts this away, but even with barebones requests you should do:

```suggestion
        response = requests.post(url, headers=self._auth_header, json=body)
```